### PR TITLE
Fix gulp ignores

### LINKED ISSUE #452: 'Ignore' setting doesn't work
Hi,

Straight to the point.
I created simple files structure:
zrzut ekranu 2015-12-13 23 34 29

Then I copied .sass-lint.yml from here: https://github.com/sasstools/sass-lint/blob/develop/lib/config/sass-lint.yml

And modified it by adding 'ignore' setting:
zrzut ekranu 2015-12-13 23 34 50

My gulpfile looks like this:
zrzut ekranu 2015-12-13 23 40 45

When i run gulp it lints every .scss file in my project, and it doesn't ignore files I set in 'ignore' setting.

### PR DESCRIPTION
**Pull request by DanPurdy**
Part 1 regarding the fixing of ignores from the gulp-sass-lint plugin the second part will follow in the gulp-sass-lint repo

closes #451
fixes #452 

`<DCO 1.1 Signed-off-by: Dan Purdy dan@danpurdy.co.uk>`

### CODE REVIEW

**Comment by DanPurdy** [comment]
Part 2 of this fix for the gulp-sass-lint plugin is at https://github.com/sasstools/gulp-sass-lint/pull/42

File: index.js [hunk 1]
```diff
@@ -144,6 +145,31 @@ sassLint.lintText = function (file, options, configPath) {
 };
 
 /**
+ * Handles ignored files for plugins such as the gulp plugin. Checks every file passed to it against
+ * the ignores as specified in our users config or passed in options.
+ *
+ * @param {object} file - The file/text to be linted
```
Comment (bgriffith) [inline index.js]: Capital `O` for jsdoc object type

File: index.js [hunk 2]
```diff
@@ -144,6 +145,31 @@ sassLint.lintText = function (file, options, configPath) {
...
+ * @param {object} options - The user defined options directly passed in
+ * @param {object} configPath - Path to a config file
+ * @returns {object} Return the results of lintText - a results object
+ */
+sassLint.lintFileText = function (file, options, configPath) {
+  var config = this.getConfig(options, configPath);
+  var ignores = config.files ? config.files.ignore : [];
```
Comment (bgriffith) [inline index.js]: var config = this.getConfig(options, configPath),
    ignores = config.files ? config.files.ignore : [];

File: tests/main.js [hunk 3]
```diff
@@ -53,95 +55,165 @@ var resultsObj = [{
...
   //////////////////////////////
-  it('should equal 3 warnings', function (done) {
-    var result = lint.warningCount(resultsObj);
 
-    assert.equal(3, result.count);
-    done();
-  });
+  describe('lintFileText', function () {
+    var file;
+    var fileObj;
```
Comment (bgriffith) [inline tests/main.js]:
```js
var file,
    fileObj;
```

**Comment by bgriffith** [comment]
Few tiny format changes, code logic looks good :shipit:

File: index.js (hunk 1 above)
Comment (DanPurdy) [inline index.js]: The rest of the file is using lower case o's so I wont update this one to be different.

**Comment by DanPurdy** [comment]
Removing the approved flag for now as this had merge conflicts which ended in me manually editing the tests/main.js file so it probably needs an extra review again just to be sure. I've verified all the tests are there.

**Pull request merged**